homeassistant/components/ondilo_ico/sensor.py

```diff
@@ -48,14 +48,20 @@
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Ondilo ICO sensors."""
 
-    api = hass.data[DOMAIN][entry.entry_id]
+    def get_all_pools_data():
+        """Fetch pools and add pool details and last measures to pool data."""
+        api = hass.data[DOMAIN][entry.entry_id]
```

Comment:
Accept the api as a parameter instead. Then we can move this function out to the module level. Flat is better than nested. 👍

Reply:
ok done.